src/Cli/Microsoft.TemplateEngine.Cli/Commands/InteractiveMode/InteractiveMode.cs

```diff
@@ -0,0 +1,253 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.CommandLine.Parsing;
+using Microsoft.TemplateEngine.Abstractions;
+using Microsoft.TemplateEngine.Edge;
+using Microsoft.TemplateEngine.Edge.Settings;
+
+namespace Microsoft.TemplateEngine.Cli.Commands.InteractiveMode
+{
+    public static class InteractiveMode
+    {
+        public static async Task EnterInteractiveMode(InvocationContext context, Func<InvocationContext, Task> next)
+        {
+            // TODO: make sure we have access to Instantiate command
+            if (context.ParseResult.CommandResult.Command.Name == "new" || context.ParseResult.CommandResult.Command.Name == "create")
+            {
+                InstantiateCommand? instantiateCommand = GetInstantiateCommand(context.ParseResult.CommandResult.Command);
+                if (instantiateCommand is null)
+                {
+                    return;
+                }
+
+                var enterInteractiveMode = ShouldEnterInteractiveMode(context.ParseResult);
+                if (enterInteractiveMode)
+                {
+                    // TODO: add the default skip key
+                    // TODO: short name validation. right now we are assuming that the provided template exists and is valid
+                    context.Console.WriteLine("Currently under development, not all features implemented yet");
+
+                    // --- Workaround to get the environmental settings at this stage. Firgure this out later
+                    ITemplateEngineHost host = instantiateCommand.Host(context.ParseResult);
```

Comment:
all of this should be done in `InstantiateCommand` class instead.

the method should return the collection of questions to ask, possible options, and how to add them to current command.

ideally this should be queried each time after the answer is received to
- validate it
- recreate the next questions as they may depend on choices made. Example: if user selects F# language, some of parameters might not be available or have different values.

Reply:
Moved things to the `InstantiateCommand` class. At this level of prototype I want to limit the amount of choices that our / parameter evaluations do. I feel like it is beyond this scope and it is valuable to sit down and plan this work and some decisions beforehand.

I have also limited this to strings right now, as I was not sure how to use generic types. I think it is good enough for a rough prototype, but it should definitely be the next step for this.